CMakeLists.txt

```diff
@@ -82,6 +82,23 @@ endif()
 target_link_libraries(llvm-dialects-tblgen PRIVATE llvm_dialects_tablegen)
 set_compiler_options(llvm-dialects-tblgen)
 
+#if __ANDROID__
+if(ANDROID)
+    if (ANDROID_ABI MATCHES "x86_64")
+        target_link_options(llvm-dialects-tblgen PRIVATE
+            LINKER:--rpath,$ENV{ANDROID_PRODUCT_OUT}/system/lib64/bootstrap/:$ENV{ANDROID_PRODUCT_OUT}/system/lib64/
+            LINKER:-dynamic-linker,$ENV{ANDROID_PRODUCT_OUT}/apex/com.android.runtime/bin/linker64
+        )
+    elseif(ANDROID_ABI MATCHES "x86")
+        # There is a limitation here. The PID should less than 65535 for 32-bit executable.
+        target_link_options(llvm-dialects-tblgen PRIVATE
+            LINKER:--rpath,$ENV{ANDROID_PRODUCT_OUT}/system/lib/bootstrap/:$ENV{ANDROID_PRODUCT_OUT}/system/lib/
+            LINKER:-dynamic-linker,$ENV{ANDROID_PRODUCT_OUT}/apex/com.android.runtime/bin/linker
+        )
+    endif()
```

Comment:
I know very little about cross compilation, but it seems a bit weird to teach a generic tool like LLVM dialects about specifics of Android builds. How does that work for other tools, e.g. as part of LLVM?

Also, depending on environment variables feels a bit weird -- aren't these available as CMake variables?

Reply:
To my knowledge, this is the only place where we've had to use the Android linker (which typically gets installed on the Android system itself). The initial solution was to copy the linker to /system/bin/linker(64), as that's where the linker is expected on the Android system. However, copying Android binaries (and libraries) into the root system of the Linux build machine is hacky.

I don't believe we have a CMake variable necessarily pointing to the out directory, as this is the only item that has to be retrieved from the AOSP build output. The remaining AOSP dependencies can be fetched from the prebuilt VNDK.